akka-stream/src/main/scala/akka/stream/impl/io/InputStreamPublisher.scala

```diff
@@ -87,8 +87,6 @@ private[akka] class InputStreamPublisher(is: InputStream, bytesReadPromise: Prom
 
   /** BLOCKING I/O READ */
   def loadChunk() = try {
-    val arr = Array.ofDim[Byte](chunkSize)
```

Comment:
As for the code:

~~~scala
if (readBytes == chunkSize) availableChunks :+= ByteString1C(arr)
         else availableChunks :+= ByteString1C(arr).take(readBytes)
~~~

I'd probably write it as:

~~~scala
availableChunks :+= ByteString1C(arr).take(readBytes)
~~~

Because a `take` of the full length (i.e. `readBytes` == `chunkSize`) returns the ByteString instance itself (no-op), and anything where `readBytes` is smaller than `chunkSize` will do the right thing.